tests/testnet/network.go

```diff
@@ -52,8 +56,8 @@ var _ interfaces.Network = &testNetwork{}
 
 type testNetwork struct {
 	teleporterContractAddress common.Address
-	primaryNetwork            interfaces.SubnetTestInfo
-	subnets                   []interfaces.SubnetTestInfo
+	primaryNetwork            *interfaces.SubnetTestInfo
+	subnetsInfo               map[ids.ID]*interfaces.SubnetTestInfo
```

Comment:
I'm fairly ambivalent, but I think keeping this is a slice and just implementing a getter method on the `testNetwork` struct that takes an `ids.ID` and returns a `SubnetTestInfo` or nil would be slightly cleaner.